src/components/RichSelect/index.tsx

```diff
@@ -380,96 +469,89 @@ const ValueContainer = ({
   </components.ValueContainer>
 )
 
-ValueContainer.defaultProps = {
-  children: null,
-  hasValue: false,
-  isDisabled: false,
-  isMulti: false,
-  noTopLabel: false,
-  selectProps: {},
-}
 ValueContainer.propTypes = {
   children: PropTypes.node,
-  hasValue: PropTypes.bool,
-  isDisabled: PropTypes.bool,
-  isMulti: PropTypes.bool,
-  noTopLabel: PropTypes.bool,
-  selectProps: SelectContainer.propTypes.selectProps,
+  hasValue: PropTypes.bool.isRequired,
+  isDisabled: PropTypes.bool.isRequired,
+  isMulti: PropTypes.bool.isRequired,
+  selectProps: SelectContainer.propTypes?.selectProps,
 }
 
-const inputStyles = ({ hasValue, noTopLabel, placeholder }) => css`
+const inputStyles = ({
+  isMulti,
+  hasValue,
+  noTopLabel,
+  placeholder,
+}: Partial<SelectProps>) => css`
   padding-top: ${(placeholder && hasValue) || noTopLabel ? 11 : 0}px;
   margin-left: 0px;
+  ${!isMulti && 'caret-color: transparent'};
 `
 
-const Input = ({
-  isMulti,
-  hasValue,
-  selectProps: { inputId, labelId, noTopLabel, placeholder, ...selectProps },
+const Input: FunctionComponent<InputProps<SelectOption> & WithSelectProps> = ({
+  isMulti = false,
+  hasValue = false,
+  selectProps: {
+    inputId,
+    labelId,
+    noTopLabel,
+    placeholder,
+    ...selectProps
+  } = {},
   ...props
 }) => (
   <components.Input
     {...props}
-    css={inputStyles({ hasValue, noTopLabel, placeholder })}
-    style={{
-      caretColor: !isMulti && 'transparent',
-    }}
+    css={inputStyles({ hasValue, isMulti, noTopLabel, placeholder })}
     id={inputId}
     aria-controls={labelId}
-    selectProps={{ ...selectProps, placeholder }}
+    hasValue={hasValue}
+    isMulti={isMulti}
+    selectProps={
+      { ...selectProps, placeholder } as InputProps<SelectOption>['selectProps']
+    }
   />
 )
+
 Input.propTypes = {
-  hasValue: PropTypes.bool,
-  isMulti: PropTypes.bool,
-  selectProps: SelectContainer.propTypes.selectProps,
+  hasValue: PropTypes.bool.isRequired,
+  isMulti: PropTypes.bool.isRequired,
```

Comment:
Why is it required if it has a default value ?

```js
isMulti = false,
hasValue = false,
```

Reply:
I removed default values, it's required by react-select typing of components.Input `InputProps`.